### PR TITLE
Global Styles Cleanup Act 2

### PR DESCRIPTION
**Pull request by kyleknighted**
## Description

Continuation of https://github.com/Graylog2/graylog2-server/pull/7402

Clearing out the `GlobalThemeStyles` file and moving specific styles to their appropriate components.

## Motivation and Context

Clear out what gets rendered on every page

_Existing eslint errors stayed or else this would have been an even larger PR_

## How Has This Been Tested?

Using built in snapshot tests and clicking through the app

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-web-interface/src/components/graylog/Modal.jsx [hunk 1]
```diff
@@ -16,6 +16,27 @@ const Modal = styled(BootstrapModal)(({ theme }) => css`
...
+  .modal-backdrop {
+    height: 100000%;  /* yes, really. this fixes the backdrop being cut off when the page is scrolled. */
+    z-index: 1030;
+  }
+
+  form {
+    margin-bottom: 0;
+  }
+
+  .modal-body .form-group {
```
Comment (edmundoa) [inline graylog2-web-interface/src/components/graylog/Modal.jsx]: Can we delete this from the global styles? https://github.com/Graylog2/graylog2-server/blob/f0bb3dc08c0b1ec64445c6eea15f04351cdd50fd/graylog2-web-interface/src/theme/GlobalThemeStyles.jsx#L1459

File: graylog2-web-interface/src/components/inputs/InputsList.jsx [hunk 2]
```diff
@@ -20,6 +22,30 @@ const InputTypesActions = ActionsProvider.getActions('InputTypes');
 const InputsStore = StoreProvider.getStore('Inputs');
 const SingleNodeStore = StoreProvider.getStore('SingleNode');
 
+const InputListRow = styled(Row)`
```
Comment (edmundoa) [inline graylog2-web-interface/src/components/inputs/InputsList.jsx]: I guess we can also remove some of these styles from the global styles:
- https://github.com/Graylog2/graylog2-server/blob/f0bb3dc08c0b1ec64445c6eea15f04351cdd50fd/graylog2-web-interface/src/theme/GlobalThemeStyles.jsx#L215-L221
- https://github.com/Graylog2/graylog2-server/blob/f0bb3dc08c0b1ec64445c6eea15f04351cdd50fd/graylog2-web-interface/src/theme/GlobalThemeStyles.jsx#L355-L367

File: graylog2-web-interface/src/components/notifications/Notification.jsx [hunk 3]
```diff
@@ -10,6 +11,27 @@ import ActionsProvider from 'injection/ActionsProvider';
 
 const NotificationsActions = ActionsProvider.getActions('Notifications');
 
+const StyledAlert = styled(Alert)`
```
Comment (edmundoa) [inline graylog2-web-interface/src/components/notifications/Notification.jsx]: Same here: https://github.com/Graylog2/graylog2-server/blob/f0bb3dc08c0b1ec64445c6eea15f04351cdd50fd/graylog2-web-interface/src/theme/GlobalThemeStyles.jsx#L195-L205

File: graylog2-web-interface/src/components/navigation/InactiveNavItem.jsx [hunk 4]
```diff
@@ -4,21 +4,19 @@ import { NavItem } from 'components/graylog';
...
 
 InactiveNavItem.propTypes = {
   active: PropTypes.any,
-  children: PropTypes.oneOfType([
-    PropTypes.array,
-    PropTypes.element,
-    PropTypes.string,
-  ]).isRequired,
+  children: PropTypes.oneOfType([PropTypes.array, PropTypes.element, PropTypes.string]).isRequired,
+  className: PropTypes.string,
```
Comment (edmundoa) [inline graylog2-web-interface/src/components/navigation/InactiveNavItem.jsx]: Could you please explain why we made this change here? I don't see where it's being used.

File: graylog2-web-interface/src/components/systemmessages/SystemMessagesList.jsx [hunk 5]
```diff
@@ -1,31 +1,43 @@
...
-}
+`;
+
+const TimestampTH = styled.th`
+  width: 200px;
+`;
+
+const SystemMessagesList = ({ messages }) => {
+  return (
+    <SystemMessagesTable className="system-messages" striped hover condensed>
```
Comment (edmundoa) [inline graylog2-web-interface/src/components/systemmessages/SystemMessagesList.jsx]: I think we don't need the `className` here any more, and same goes with the style definition in the global styles.

File: graylog2-web-interface/src/views/components/datatable/DataTable.jsx [hunk 6]
```diff
@@ -100,20 +101,18 @@ const DataTable = ({ config, currentView, data, fields }: Props) => {
...
-                       columnPivots={columnPivots}
-                       fields={fields}
-                       rollup={rollup}
-                       rowPivots={rowPivots}
-                       series={series} />
-            </thead>
-            {formattedRows}
-          </table>
-        </div>
+        <MessagesTable>
```
Comment (edmundoa) [inline graylog2-web-interface/src/views/components/datatable/DataTable.jsx]: I think this style definition can also be removed:
https://github.com/Graylog2/graylog2-server/blob/f0bb3dc08c0b1ec64445c6eea15f04351cdd50fd/graylog2-web-interface/src/theme/GlobalThemeStyles.jsx#L1279-L1282

File: graylog2-web-interface/src/views/components/datatable/MessagesTable.jsx [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+  }
+
+  tr.message-row .message-wrapper {
+    line-height: 1.5em;
+    white-space: pre-line;
+    max-height: 6em; /* show 4 lines: line-height * 4 */
+    overflow: hidden;
+  }
+
+  tr.message-row .message-wrapper:after {
```
Comment (edmundoa) [inline graylog2-web-interface/src/views/components/datatable/MessagesTable.jsx]: Stylelint complains about this:
```
Expected double colon pseudo-element notation (selector-pseudo-element-colon-notation)
```

File: graylog2-web-interface/src/views/components/datatable/MessagesTable.jsx [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+
+  tr.message-detail-row .row {
+    margin-right: 0;
+  }
+
+  tr.message-detail-row div[class*="col-"] {
+    padding-right: 0;
+  }
+
+  th i.sort-order-desc {
```
Comment (edmundoa) [inline graylog2-web-interface/src/views/components/datatable/MessagesTable.jsx]: Do we need these styles in the global styles?
https://github.com/Graylog2/graylog2-server/blob/f0bb3dc08c0b1ec64445c6eea15f04351cdd50fd/graylog2-web-interface/src/theme/GlobalThemeStyles.jsx#L776-L789

File: graylog2-web-interface/src/components/navigation/InactiveNavItem.jsx (hunk 4 above)
Comment (kyleknighted) [reply graylog2-web-interface/src/components/navigation/InactiveNavItem.jsx]: The gist is that styled-components passes its styles via `className` so in order to do something like `styled(Component)` that Component needs to accept the `className` prop

https://styled-components.com/docs/basics#styling-any-component

Comment (kyleknighted) [reply graylog2-web-interface/src/components/navigation/InactiveNavItem.jsx]: >  I don't see where it's being used.

`src/components/navigation/NotificationBadge.jsx:14`

Comment (edmundoa) [reply graylog2-web-interface/src/components/navigation/InactiveNavItem.jsx]: Ok, it makes sense now, thanks!

**Pull request merged**